include/ddc/kernels/splines/matrix_maker.hpp

```diff
@@ -11,9 +11,30 @@
 
 namespace ddc::detail {
 
+/**
+ * @brief A static factory for Matrix classes.
+ *
+ * At the moment only the sparse matrix is available.
+ */
 class MatrixMaker
 {
 public:
+    /**
+     * @brief Construct a sparse matrix
+     *
+     * @tparam the Kokkos::ExecutionSpace on which matrix-related operation will be performed.
+     * @param n The size of one of the dimensions of the matrix (which is necessarily squared).
+     * @param cols_per_chunk A hyperparameter used by the slicer (internal to the solver) to define the size
+     * of a chunk of right-hand-sides of the linear problem to be computed in parallel (chunks are treated
+     * by the linear solver one-after-the-other).
+     * This value is optional. If no value is provided then the default value is chosen by the requested solver.
```

Comment:
```suggestion
     * @param cols_per_chunk A parameter used by the slicer (internal to the solver) to define the size
     * of a chunk of right-hand sides of the linear problem to be computed in parallel (chunks are treated
     * by the linear solver one-after-the-other).
     * This value is optional. If no value is provided then the default value is chosen by the requested solver.
```

What distinction between parameter and hyperparameter do we want to make ?

Reply:
Maybe this terminology is too specific to ML. Hyperparameters are parameters of the training, not of the model.

If we do the parallel with linear solvers, it would be a parameter of the solver, not of the linear problem.